.github/workflows/update_api.yml

```diff
@@ -0,0 +1,60 @@
+name: Update API 
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "30 3 * * *"
+jobs:
+  update-api:
+    if: ${{ github.repository == 'opensearch-project/opensearch-py' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - name: Config git to rebase
+        run: git config --global pull.rebase true
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+      - name: Install dependencies
+        run: |
+          python3.7 -m pip install nox
+      - name: Generate API
+        run: nox -s generate
+      - name: Update CHANGELOG
+        uses: jacobtomlinson/gha-find-replace@v3
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        with:
+          find: "- Your contribution here."
+          replace: "- Updated opensearch-py to reflect the latest OpenSearch API spec.\n- Your contribution here."
```

Comment:
Let's change these messages to be something more specific? e.g. the commit of opensearch-api-specification that was used? This way we don't have a thousand PRs/commit messages that have the same title?